mod/state-transition/pkg/core/state_processor_staking.go

```
@@ -289,24 +310,23 @@
-	// Update the next validator index to start the next withdrawal sweep
-	//#nosec:G701 // won't overflow in practice.
-	if numWithdrawals == int(sp.cs.MaxWithdrawalsPerPayload()) {
-		// Next sweep starts after the latest withdrawal's validator index
-		nextValidatorIndex =
-			(expectedWithdrawals[len(expectedWithdrawals)-1].GetIndex() + 1) %
-				math.U64(totalValidators)
+	// Update the next validator index to start the next withdrawal sweep.
+	var nextValIndex math.ValidatorIndex
+	if numWithdrawals > 1 &&
+		//#nosec:G701 // won't overflow in practice.
+		numWithdrawals == int(sp.cs.MaxWithdrawalsPerPayload()) {
```

golangci-lint

[high] G115: integer overflow conversion uint64 -> int

(gosec)